experimental/list_formatter/src/provider.rs

```diff
@@ -218,28 +225,36 @@ impl<'data> From<ListJoinerPattern<'data>> for ConditionalListJoinerPattern<'dat
 }
 
 impl<'a> ConditionalListJoinerPattern<'a> {
-    pub fn parts(&'a self, following_value: &str) -> PatternParts<'a> {
-        match &self.special_case {
-            Some(SpecialCasePattern { condition, pattern }) if condition.test(following_value) => {
-                pattern.borrow_tuple()
+    /// Returns the pattern parts, and if materialized, the following value as a string
+    pub fn parts<W: Writeable>(
+        &'a self,
+        following_value: &W,
+    ) -> (PatternParts<'a>, Option<alloc::string::String>) {
+        match (&self.special_case, following_value.peek()) {
+            (Some(SpecialCasePattern { condition, pattern }), None) => {
+                let v = following_value.writeable_to_string();
```

Comment:
Suggestion: Another way to do this, which avoids the extra allocation and all the spaghetti that goes along with it, is to write the "normal" pattern to the FormattedString, then write the following_value, then check the condition, and if the condition matches, use `splice` to replace the old literal with the replacement literal.

Reply:
If we want to produce a `Writeable` we don't have `slice`, we can only append.